services/searcher.ts

```diff
@@ -458,6 +459,60 @@ class Searcher {
     };
   }
 
+  filterSection(
+    curSection: Section,
+    honorsFilter: boolean,
+    campusFilter: string
+  ): boolean {
+    if (honorsFilter && !curSection.honors) {
+      return false;
+    }
+    if (campusFilter !== "" && curSection.campus !== campusFilter) {
+      return false;
+    }
+
+    return true;
+  }
+
+  filterResults(filters, results): SearchResult[] {
+    // Do extra filtering if Honors or Campus Filters are being applied since those filter sections,
+    // which ElasticSearch doesn't actually filter out.
```

Comment:
```suggestion
  /** 
   * Perform extra filtering if Honors or Campus Filters are being applied.
   * Elasticsearch will return ALL sections for a class if ANY of its sections match the filter.
   * eg. if a class has 100 sections and 1 of them is Honors, applying the Honors filter
   *       in Elasticsearch will return all 100 sections.
   * This function only returns the sections which match the filter.
   */
  filterResults(filters, results): SearchResult[] {
```

For function comments, we prefer [JSDoc](https://jsdoc.app/) - that way, when someone hovers over the function name, they'll see the comment. (I know the majority of our codebase doesn't use that, but we want to write better code going forward & slowly refactor the old code to match)